uc.c

```diff
@@ -1144,6 +1144,34 @@ uint32_t uc_mem_regions(uc_engine *uc, uc_mem_region **regions, uint32_t *count)
 }
 
 UNICORN_EXPORT
+uc_err uc_mmio_map(uc_engine *uc, uint64_t address, size_t size,
+        uc_cb_mmio_t callback, void *user_data)
+{
+    uc_err res;
+    struct mmio_data *data;
+
+    if (callback == NULL)
+        return UC_ERR_ARG;
+
+    if (uc->mem_redirect) {
+        address = uc->mem_redirect(address);
+    }
+
+    res = mem_map_check(uc, address, size, UC_PROT_ALL);
+    if (res)
+        return res;
+
+    data = g_new(struct mmio_data, 1);
```

Comment:
are you relying on uc_mem_unmap() to free the resource for you? if then, then uc_mem_unmap() must be aware of this allocated memory to release that.